connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -129,6 +129,9 @@ message SQL {
   // (Optional) A sequence of expressions for positional parameters in the SQL query text.
   // It cannot coexist with `named_arguments`.
   repeated Expression pos_arguments = 5;
+
```

Comment:
I'd like to introduce a slightly more generic mechanism here. Where we define a CTE or something similar. This way we can fix recursion limits, and we can reduce tree duplication.

Reply:
Sounds good, let me mark this PR wip for now